genesis/process/shardGenesisBlockCreator.go

```diff
@@ -361,19 +362,34 @@ func createProcessorsForShardGenesisBlock(arg ArgsGenesisBlockCreator, enableEpo
 		return nil, err
 	}
 
+	convertedCrossChainAddresses, err := mainFactory.DecodeAddresses(
+		arg.Core.AddressPubKeyConverter(),
+		arg.SystemSCConfig.ESDTSystemSCConfig.WhiteListedCrossChainMintAddresses,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	mapCrossChainAddresses := make(map[string]struct{})
+	for _, address := range convertedCrossChainAddresses {
```

Comment:
you have this code in a lot of places. I think it would be better to move it inside the builtInFunctionFactory.

Reply:
Refactored